docs/web/docs/explanations/architect_api.md

```diff
@@ -0,0 +1,108 @@
+# Architects, Routers, and `mephisto-task`: The Architect API.
+
+The Architect API is what allows Mephisto to completely abstract away the process of getting workers to operate in Mephisto tasks. In short, it aims to cover 4 primary functions:
+1. `Worker`/`Agent` registration and validation
+2. Seamless frontend access to `Unit` data made available on the Python side both when accepting and working on a task.
+3. Method to submit completed task data to the backend when a task is complete.
+4. Worker liveliness and status checking and syncing.
+
+We also have a few additional goals, which influence design decisions:
+1. Tasks without the _need_ for live interaction with the backend should be REST-ful rather than socket-based. This prevents the need to be fully connected throughout the task, reducing unexpected disconnects.
+2. The system should include failsafes to allow for retries.
+3. We should prioritize low latency where possible.
+4. The interaction layer should be simple. 
+
+This document aims to describe how we got from these requirements and goals to the system we have now.
+
+## Requirements
+
+### `Worker`/`Agent` registration
+Mephisto allows for multiple stages of filtering for a worker, which leads the registration process to be staged. The complete order is listed below:
+0. `CrowdProvider`-set qualifications may prevent a worker from even seeing a task.
+1. `Qualification`s may be set that a `CrowdProvider` is not aware of. Workers not meeting these qualifications should be filtered.
```

Comment:
This is the only one that isn't clear to me. What is meant by "should be filtered" - by whom? Does Mephisto take care of this, or is it up to the task author? What is a use case where you'd do this over the option above?

Reply:
Ah Mephisto takes care of filtering out workers by any qualifications that the task owner sets, regardless of if those qualifications are propagated to a `CrowdProvider`. I imagine this is a fairly uncommon use case, but if someone wants to keep _private qualifications_ of some kind, this would be the way to do it.

Reply:
Adding some clarifying text:
`Qualification`s may be set locally that a `CrowdProvider` is not aware of. Workers not meeting these Mephisto qualifications (as set by `SharedState.qualifications`) should be filtered out.

Reply:
Ah I see, this bit was the most helpful for me: "I imagine this is a fairly uncommon use case, but if someone wants to keep private qualifications of some kind, this would be the way to do it"